homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,272 @@
+"""Config flow to configure the Nuki integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any, Final
+
+from pynanoleaf import InvalidToken, Nanoleaf, NotAuthorizingNewTokens, Unavailable
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MODE, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.util.json import load_json, save_json
+
+from .const import DOMAIN
+from .util import pynanoleaf_get_info
+
+_LOGGER = logging.getLogger(__name__)
+
+# For discovery integration import
+CONFIG_FILE: Final = ".nanoleaf.conf"
+
+CONF_AUTO: Final = "auto"
+CONF_MANUAL: Final = "manual"
+
+CONF_STEP_USER: Final = "user"
+CONF_STEP_AUTO: Final = "auto"
+CONF_STEP_MANUAL: Final = "manual"
+
+SIMPLE_USER_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+ADVANCED_USER_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_MODE, default=CONF_AUTO): vol.In(
+            {
+                CONF_AUTO: "Request a new token (recommended)",
+                CONF_MANUAL: "Enter a token manually",
```

Comment:
Why would we even offer this option? Seems like it's a developer option ? Let's remove it, or if you have a good use case, let's only show it for advanced users?

Reply:
So I just read https://www.home-assistant.io/integrations/nanoleaf/#getting-the-auth-token. Definitely drop offering users to enter a token.